src/ar65/library.c

```diff
@@ -249,9 +249,10 @@ void LibOpen (const char* Name, int MustExist, int NeedTemp)
     if (NeedTemp) {
 
         /* Create the temporary library name */
-        NewLibName = xmalloc (strlen (Name) + strlen (".temp") + 1);
+        NewLibName = xmalloc (strlen (Name) + strlen (".temp-") + 8 + 1);
         strcpy (NewLibName, Name);
-        strcat (NewLibName, ".temp");
+        strcat (NewLibName, ".temp-");
+        sprintf (NewLibName + strlen (NewLibName), "%X", (unsigned int)getpid());
```

Comment:
Why don't you use something like mkstemp(3)?

Reply:
Might be a good idea. But I was lazy and didn't want to bother about the issue that the temp file could be in another directory and file system than the output file.

Reply:
> The right solution to fix this problem is to fix the geos-apple target.

Where exactly is this happening? I don't find it.

Reply:
See PR #1405.

Reply:
Unfortunately, I had to retract my PR #1405, as the fix generated an invalid apple2enh library.

Reply:
So... is this PR still relevant, or should I retract is as well?

Reply:
Apart from the question if we can/will fix the root cause of the issue at hand (apple2enh lib) I'm generally following the idea to add `getpid ()` to the temp file name.

However, I'm not that happy with the currently proposed code:

* When calling the heavyweight sprintf then we should make it do all the work. Something like `sprintf (NewLibName, "%s.%x.temp", Name, (unsigned int)getpid());`
* I pretty much dislike "magic constants" in the source code - like the `8` you introduced. Additionally I believe the code might produce a memory overwrite issue when `(unsigned int)` becomes 64-bit. As far as I understand replacing `8` with `2 * sizeof (unsigned int)` should take care of both aspects.

Reply:
You are right. It was a "quick hack". I've updated the PR.